### DISCUSSION TITLE
Remove CSV export limitation

### DISCUSSION THREAD

**Discussion by sidiyo** [wazuh/wazuh#17288] [General]
Regards,

Is there any way to remove the limitation to export the CSV of the events?

Currently there is a limitation for only 10k of events per CSV, this ends up being extremely laborious when it comes to logs..., where it easily exceeds this, causing the user to export several CSV files to later centralize.

Thank you for every future reply!

**Comment by sidiyo** [comment]
Below is evidence of the limitation, since when we load it into excel, it loads 10k of logs.

No, the limitation is not in excel, I already validated this, I even tried to open it in Power BI, and the same limitation remained.

![image](https://github.com/wazuh/wazuh/assets/40663746/f2ef5c1f-6166-491b-b733-5e884c545b96)

**Comment by s-ocando** [comment]
Hi @sidiyo,

Unfortunately, the 10,000 CSV row limit is **non-configurable**, as stated in the [Creating reports with the Dashboards interface](https://opensearch.org/docs/2.6/dashboards/reporting/) documentation.

To overcome this limitation, consider creating a simple script to retrieve the information through API calls. Check the available methods to paginate search results: [Paginate results](https://opensearch.org/docs/2.6/search-plugins/searching-data/paginate/).

Note that for deep pagination, it's recommended to use [Point in Time](https://opensearch.org/docs/2.6/search-plugins/point-in-time/) and the `search_after` parameter. 

I hope you find this information useful. Let us know if you have any further questions.
